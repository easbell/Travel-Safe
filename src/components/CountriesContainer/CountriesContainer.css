.container {
  height: 800px;
  overflow: scroll;
  margin-top: 48px;
  display: grid;
  grid-template-rows: 150px;
  grid-gap: 30px;
  padding: 0px 35px;
  width: 75%;
  margin: 48px auto;
  grid-template-columns: repeat( auto-fit, minmax(350px, 1fr) );
}

.react-fuzzy-filter__input {
  width: 25%;
  border: 0;
  box-shadow: 2px 2px 11px #818181;
  border-radius: 5px;
  height: 49px;
  padding-left: 15px;
  font-size: 20px;
}