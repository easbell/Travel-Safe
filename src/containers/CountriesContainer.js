import React, { Component } from 'react';
import { connect } from 'react-redux';
import { combineData } from '../helpers/combineData';

export class CountriesContainer extends Component {
  cleanData = () => {
    const { safety, vaccines } = this.props;
    const cleaned = combineData(safety, vaccines);
    console.log(cleaned)
  }

  render() {
    return (
      <div>
        {this.cleanData()}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  safety: state.safety,
  vaccines: state.vaccines
});

export default connect(mapStateToProps)(CountriesContainer);