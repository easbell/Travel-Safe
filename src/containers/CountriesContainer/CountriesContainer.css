.container {
  height: 800px;
  overflow: scroll;
}