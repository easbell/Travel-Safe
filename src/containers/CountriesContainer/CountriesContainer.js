import React, { Component } from 'react';
import './CountriesContainer.css';
import Country from '../Country/Country';
import fuzzyFilterFactory from "react-fuzzy-filter";
const { InputFilter, FilterResults } = fuzzyFilterFactory();

export class CountriesContainer extends Component {
  renderCountries = (filtered) => {
    return filtered.map(country => {
      return <Country key={country.name} {...country}/>
    });
  }

  render() {
    const { countries } = this.props;
    const fuseConfig = {
      keys: ['name'],
      matchAllTokens: true,
      tokenize: true,
      threshold: 0
    };

    return (
      <div>
        {countries.length === 57 && <InputFilter />}
        <FilterResults items={countries} fuseConfig={fuseConfig}>
          {filteredItems => {
            return (
              <div className='container'>
                {this.renderCountries(filteredItems)}
              </div>
            )
          }}
        </FilterResults>
      </div>
    )
  }
}

export default CountriesContainer;