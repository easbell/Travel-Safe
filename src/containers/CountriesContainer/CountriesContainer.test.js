import React from 'react';
import { CountriesContainer, mapStateToProps } from './CountriesContainer';
import { shallow } from 'enzyme';

describe('CountriesContainer', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(
      <CountriesContainer />)
      expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with the vaccine object', () => {
      const mockState = {
        vaccines: {country: 'France', vaccines: []},
        other: 'other'
      }
      const expected = {
        vaccines: {country: 'France', vaccines: []}
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});