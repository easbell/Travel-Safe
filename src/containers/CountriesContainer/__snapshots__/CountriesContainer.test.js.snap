// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesContainer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CountriesContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InputFilter
          classPrefix="react-fuzzy-filter"
          debounceTime={0}
          inputProps={Object {}}
          onChange={[Function]}
        />,
        <FilterResults
          defaultAllItems={true}
          fuseConfig={
            Object {
              "keys": Array [
                "name",
              ],
              "matchAllTokens": true,
              "threshold": 0,
              "tokenize": true,
            }
          }
          prefilters={Array []}
        >
          [Function]
        </FilterResults>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "classPrefix": "react-fuzzy-filter",
          "debounceTime": 0,
          "inputProps": Object {},
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": [Function],
          "defaultAllItems": true,
          "fuseConfig": Object {
            "keys": Array [
              "name",
            ],
            "matchAllTokens": true,
            "threshold": 0,
            "tokenize": true,
          },
          "items": undefined,
          "prefilters": Array [],
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InputFilter
            classPrefix="react-fuzzy-filter"
            debounceTime={0}
            inputProps={Object {}}
            onChange={[Function]}
          />,
          <FilterResults
            defaultAllItems={true}
            fuseConfig={
              Object {
                "keys": Array [
                  "name",
                ],
                "matchAllTokens": true,
                "threshold": 0,
                "tokenize": true,
              }
            }
            prefilters={Array []}
          >
            [Function]
          </FilterResults>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "classPrefix": "react-fuzzy-filter",
            "debounceTime": 0,
            "inputProps": Object {},
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": [Function],
            "defaultAllItems": true,
            "fuseConfig": Object {
              "keys": Array [
                "name",
              ],
              "matchAllTokens": true,
              "threshold": 0,
              "tokenize": true,
            },
            "items": undefined,
            "prefilters": Array [],
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
